### PR TITLE
CloudWatch Logs: Set default logs query and disable button when empty

### LINKED ISSUE #61070: Cloudwatch Logs: Prepopulate default query and disable blank queries
If a user tries to create a cloudwatch query but forgets to pass a query string they get back this:
Screenshot 2023-01-05 at 3 24 02 PM

This error doesn't really tell them what's wrong with what they've written. They may assume for example that all logs will be returned, which is not the case. 

The AWS Console has a default query string set up to prevent this confusion:
Screenshot 2023-01-05 at 3 27 06 PM

And it shows some kind of indication to the user if it is removed:
Screenshot 2023-01-05 at 3 27 15 PM

We should do something similar in Grafana

### PR DESCRIPTION
**Pull request by iwysiu**
**What is this feature?**
When the query mode is changed, have the expression default to '' (for metrics) and to valid default query (for logs). Also, deactivate the `run queries` button when the log query is empty.

**Why do we need this feature?**

Currently, users are confused when they switch to logs mode and run it, only for it to return a confusing error. This sets the same default logs query as aws log insights and stops them from running empty queries

**Which issue(s) does this PR fix?**:

Fixes #61070 

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/components/QueryHeader.tsx [hunk 1]
```diff
@@ -34,12 +35,19 @@ const QueryHeader: React.FC<Props> = ({
   const { queryMode, region } = query;
   const isMonitoringAccount = useIsMonitoringAccount(datasource.resources, query.region);
   const [regions, regionIsLoading] = useRegions(datasource);
+  const emptyLogsExpression = queryMode === 'Logs' && (isCloudWatchLogsQuery(query) ? !query.expression : false);
```
Comment (kevinwcyu) [inline public/app/plugins/datasource/cloudwatch/components/QueryHeader.tsx:38]: Can this check be just `isCloudWatchLogsQuery(query) ? !query.expression : false`? looks like `isCloudWatchLogsQuery` checks for `queryMode === 'Logs'` already.

File: public/app/plugins/datasource/cloudwatch/components/QueryHeader.test.tsx [hunk 2]
```diff
@@ -117,4 +117,93 @@ describe('QueryHeader', () => {
...
+      render(
+        <QueryHeader
+          datasource={datasource}
+          query={{ ...validMetricSearchBuilderQuery, expression: 'foo' }}
+          onChange={onChange}
+          onRunQuery={jest.fn()}
+          dataIsStale={false}
+        />
+      );
+      await waitFor(() => expect(screen.queryByText('CloudWatch Metrics')).toBeInTheDocument());
```
Comment (kevinwcyu) [inline public/app/plugins/datasource/cloudwatch/components/QueryHeader.test.tsx:135]: I think something like this should work without needing `waitFor`

```suggestion
      expect(await screen.findByText('CloudWatch Metrics')).toBeInTheDocument();
```

File: public/app/plugins/datasource/cloudwatch/components/QueryHeader.test.tsx [hunk 3]
```diff
@@ -117,4 +117,93 @@ describe('QueryHeader', () => {
...
+      await act(async () => {
+        await selectEvent.select(screen.getByLabelText('Query mode'), 'CloudWatch Logs', { container: document.body });
```
Comment (kevinwcyu) [inline public/app/plugins/datasource/cloudwatch/components/QueryHeader.test.tsx:137]: I think changing `screen.getByLabelText` to `await screen.findByLabelText`, can let you remove the `await act(async () => {` on line 136.

**Review by kevinwcyu — APPROVED** [review]

File: public/app/plugins/datasource/cloudwatch/components/QueryHeader.tsx (hunk 1 above)
Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/components/QueryHeader.tsx]: done

File: public/app/plugins/datasource/cloudwatch/components/QueryHeader.test.tsx (hunk 2 above)
Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/components/QueryHeader.test.tsx]: done

File: public/app/plugins/datasource/cloudwatch/components/QueryHeader.test.tsx (hunk 3 above)
Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/components/QueryHeader.test.tsx]: done

**Review by sarahzinger — APPROVED** [review]
🚀

**Pull request merged**